unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quest_V2/QuestReward/QuestRewardComponentView.cs

```diff
@@ -0,0 +1,58 @@
+using DCL.Quest;
+using System;
+using TMPro;
+using UIComponents.Scripts.Components;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class QuestRewardComponentView : BaseComponentView<QuestRewardComponentModel>, IQuestRewardComponentView
+{
+    [SerializeField] internal TMP_Text rewardName;
+    [SerializeField] internal TMP_Text rewardQuantity;
+    [SerializeField] internal Image typeImage;
+    [SerializeField] internal ImageComponentView nftImage;
+    [SerializeField] internal NFTTypeIconsAndColors nftTypesIcons;
+    [SerializeField] private Image backgroundImage;
+    [SerializeField] private Image rarityBackgroundImage;
+
+    public override void RefreshControl()
+    {
+        SetName(model.name);
+        SetQuantity(model.quantity);
+        SetType(model.type);
+        SetRarity(model.rarity);
+        SetImage(model.imageUri);
+    }
+
+    public void SetName(string name)
+    {
+        model.name = name;
+        rewardName.text = name;
+    }
+
+    public void SetQuantity(int quantity)
+    {
+        model.quantity = quantity;
+        rewardQuantity.text = quantity <= 999 ? $"{quantity} remaining" : ">999 remaining";
+    }
```

Comment:
You can use StringBuilder, it should allocate a bit less memory.

```suggestion
    string MESSAGE = "{0} remaining";
    StringBuilder sb = new StringBuilder();    
    public void SetQuantity(int quantity)
    {
        model.quantity = quantity;
        sb.Clear();
        sb.AppendFormat(MESSAGE, quantity);
        rewardQuantity.text = quantity <= 999 ? sb.ToString() : ">999 remaining";
    }
```

I leave it up to you to apply or not this change, it affects readability.